### DISCUSSION TITLE
Camera stream works - but no still image (Snapshot)

### DISCUSSION THREAD

**Discussion by dwdonnelly** [mrlt8/docker-wyze-bridge#1284] [Q&A] [answered]
@mrlt8   I have 2 external (battery-powered) and three internal (wall-powered) wyze cameras set up.  I am setting up the "Generic Camera" platform in HA.  I have the "stream" function working for the internal cameras and as we previously discussed, the external battery-powered cameras will update the stream when activated.  However, I can not get the snapshot/still image function to work in the generic camera platform.
  The documentation (https://github.com/mrlt8/docker-wyze-bridge/wiki/Home-Assistant) shows for the still image:  `http://127.0.0.1:8123/local/cam-name`.  Trying `http://192.168.0.182:8123/local/garage-cam.jpg` produces an error: `Unable to load valid image from still image URL (e.g. invalid host, URL or authentication failure). Review log for more info.`.

The log states:

Logger: homeassistant.components.generic.config_flow
Source: components/generic/config_flow.py:188
integration: Generic Camera (documentation, issues)
First occurred: 8:36:18 AM (1 occurrences)
Last logged: 8:36:18 AM

Error getting camera image from http://192.168.0.182/local/garage-cam.jpg: ConnectError

RTSP transport protocols - no selection.
Authentication: Basic selected
Username blank
Password blank
Frame rate 2
Verify SSL certificate checked 

I have verified the IP Address (192.168.0.182) and the Home Assistant URL (http://homeassistant.local:8123) from within HA's Terminal function and the camera name from the Docker Wyze Bridge camera information.

When I try the URL: http://192.168.0.182:8123/local/garage-camera.jpg in the web browser, I get `404: Not Found`

FYI: the stream URL is very different from the documentation sample (https://github.com/mrlt8/docker-wyze-bridge/wiki/Home-Assistant).  It is `rtsp://wb:(a really long alphanumeric code)@homeassistant.local:8554/garage-cam`.  So is the still image URL very different also?

Your help is appreciated.

**Comment by mrlt8** [comment, marked as answer]
Could you try:

```
http://127.0.0.1:5000/snapshot/garage-cam.jpg?api=<your-api-key>
```

The API key should be the same as the RTSP one. You can also find it at the bottom of the WebUI.

You could also set `WB_AUTH` to false if your HA is completely local and you don't need the added authentication.

**Comment by dwdonnelly** [comment]
@mrlt8 Ok, I used `http://127.0.0.1:5000/snapshot/garage-cam.jpg?api=<with my api key>` 
 and the setup on the generic camera platform worked.  Thanks ! !


Now, do I presume I would use this same HTTP code for all the other internal and external cameras?

And why am I using IP 127.0.0.1:5000?

Now I did test one of the other internal cameras with the following: `http://homeassistant.local:5000/snapshot/pantry-v3.jpg?api=<with my api key>`......and it to successfully completed the generic camera platform configuration.

Which makes me more confused as I thought homeassistant.local was IP 192.168.0.182.

**Reply by mrlt8** [reply]
Updated the wiki to use the snapshot url instead.

Since the docker container is hosted on the same machine as home assistant, we can just use `127.0.0.1` which is a loopback address.

**Comment by dwdonnelly** [comment]
@mrlt8 Many thanks !
